### PR TITLE
Implement router

### PR DESCRIPTION
**Pull request by BondPV**

### CODE REVIEW

**Review by Alesia-V175 — APPROVED** [review]

File: src/components/main/sortBar/FilterCatalog.ts [hunk 1]
```diff
@@ -14,31 +14,32 @@ class FilterCatalog {
     private productsDB: ProductsDB,
   ) {}
 
-  resetButtonContainer = document.querySelector(IdMap.resetButton) as HTMLElement;
-
-  filtersContainer = document.querySelector(IdMap.valueFilters) as HTMLElement;
-
-  filtersRangeContainer = document.querySelector(IdMap.rangeFilters) as HTMLElement;
-
   render() {
```
Comment (Ksarelto) [inline src/components/main/sortBar/FilterCatalog.ts]: please, add modifiers and return tupe to methods

File: src/components/main/sortBar/filters/Filters.ts [hunk 2]
```diff
@@ -224,10 +226,12 @@ class Filters {
 
   // check for activity on saved from Local Storage
   checkActiveElement(elementValue: string, element: HTMLInputElement): void {
-    const filteredCatalog = LocalStorage.savedFilters.filtersValue[this.filterName as FiltersValueType];
+    if (LocalStorage.savedFilters) {
+      const filteredCatalog = LocalStorage.savedFilters.filtersValue[this.filterName as FiltersValueType];
 
-    if (filteredCatalog.includes(elementValue)) {
-      element.checked = true;
+      if (filteredCatalog.includes(elementValue)) {
```
Comment (Ksarelto) [inline src/components/main/sortBar/filters/Filters.ts]: try to not use nested 'if' statements, u can use samething like
```
element.checked = filteredCatalog.includes(elemenntValue)
```

File: src/components/pages/cartPage.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class CartPage {
+  container: HTMLElement;
+
+  constructor(container: HTMLElement) {
+    this.container = container;
+    this.renderPage();
+  }
+
+  renderPage() {
```
Comment (Ksarelto) [inline src/components/pages/cartPage.ts]: modifiers and return types

File: src/components/pages/mainPage.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  private catalog = new Catalog(this.productsDB.getProducts());
+
+  private sortCatalog = new SortCatalog(this.catalog);
+
+  private filterCatalog = new FilterCatalog(this.catalog, this.sortCatalog, this.productsDB);
+
+  private grid = new Grid();
+
+  container: HTMLElement;
```
Comment (Ksarelto) [inline src/components/pages/mainPage.ts:8]: move declaration to constructor

File: src/components/pages/productDetailsPage.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import ProductsDB from 'database/ProductsDB';
+import { IProduct } from 'types/interfaces';
+
+class ProductDetailsPage {
+  container: HTMLElement;
+
+  private productsDB = new ProductsDB();
+
+  hash: string;
```
Comment (Ksarelto) [inline src/components/pages/productDetailsPage.ts:7]: move declaration to constructor

File: src/helpers/router/Router.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  constructor() {
+    this.location = window.location;
+    this.setPage(this.location.hash.slice(1));
+    this.hashChange();
+  }
+
+  private hashChange() {
+    window.addEventListener('hashchange', () => {
+      if (this.currentPage) this.currentPage.removePage();
```
Comment (Ksarelto) [inline src/helpers/router/Router.ts]: don't use 'if' in such way, it decrease readability, better use common type, with curly bracers

```
if (some condition) {
}
```

File: src/index.html [hunk 7]
```diff
@@ -76,8 +76,8 @@
   </header>
   <main class="main">
     <div class="container">
-      <div class="main__wrap">
-        <section class="filter-bar">
+      <div class="main__wrap" id="main-container">
+        <!-- <section class="filter-bar">
```
Comment (Ksarelto) [inline src/index.html]: why is it commented?

File: src/components/main/sortBar/FilterCatalog.ts (hunk 1 above)
Comment (BondPV) [reply src/components/main/sortBar/FilterCatalog.ts]: fixed

File: src/components/main/sortBar/filters/Filters.ts (hunk 2 above)
Comment (BondPV) [reply src/components/main/sortBar/filters/Filters.ts]: fixed

File: src/components/pages/cartPage.ts (hunk 3 above)
Comment (BondPV) [reply src/components/pages/cartPage.ts]: fixed

File: src/components/pages/mainPage.ts (hunk 4 above)
Comment (BondPV) [reply src/components/pages/mainPage.ts:8]: fixed

File: src/components/pages/productDetailsPage.ts (hunk 5 above)
Comment (BondPV) [reply src/components/pages/productDetailsPage.ts:7]: fixed

File: src/helpers/router/Router.ts (hunk 6 above)
Comment (BondPV) [reply src/helpers/router/Router.ts]: fixed

File: src/index.html (hunk 7 above)
Comment (BondPV) [reply src/index.html]: comment deleted. markup creation moved to js code

**Pull request merged**